app/ide-desktop/lib/client/watch.ts

```diff
@@ -129,10 +90,12 @@ const ELECTRON_ARGS = [path.join(IDE_DIR_PATH, 'index.cjs'), '--', ...process.ar
 
 process.on('SIGINT', () => {
     console.log('SIGINT received. Exiting.')
-    // The `esbuild` process seems to remain alive at this point and will keep our process
-    // from ending. Thus, we exit manually. It seems to terminate the child `esbuild` process
-    // as well.
-    process.exit(0)
+    void fs.rm(IDE_DIR_PATH, { recursive: true, force: true }).then(() => {
+        // The `esbuild` process seems to remain alive at this point and will keep our process
+        // from ending. Thus, we exit manually. It seems to terminate the child `esbuild` process
+        // as well.
+        process.exit(0)
+    })
```

Comment:
Why do we delete the directory now? Don't we want to keep it around for future invocations of this script?

Reply:
this is a watch script so it's probably marginally better to be sure that stray files aren't accidentally making the watch script work when actually we've deleted some very important files.

plus we use esbuild for bundling the electron app so re-building all the files should be fast enough (<<1s)